### ISSUE TITLE
ANR in onTrimMemory due to DB call on main thread

### ISSUE THREAD

**Issue by raymondctc** [Instabug/Instabug-Android#298]
Seeing some freeze issue as described in the issue title

### Steps to Reproduce the Problem
ANR dump

```
"main" prio=5 tid=1 TimedWaiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x711b0e78 self=0x72fa2b2c00
  | sysTid=13708 nice=-10 cgrp=default sched=0/0 handle=0x72fb818ed0
  | state=S schedstat=( 167357249377 40716269072 195119 ) utm=14805 stm=1930 core=0 HZ=100
  | stack=0x7fded8f000-0x7fded91000 stackSize=8192KB
  | held mutexes=
  at sun.misc.Unsafe.park(Native method)
  - waiting on an unknown object
  at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
  at android.database.sqlite.SQLiteConnectionPool.waitForConnection(SQLiteConnectionPool.java:756)
  at android.database.sqlite.SQLiteConnectionPool.acquireConnection(SQLiteConnectionPool.java:380)
  at android.database.sqlite.SQLiteSession.acquireConnection(SQLiteSession.java:896)
  at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
  at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:61)
  at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
  at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
  at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1443)
  at android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1290)
  at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1161)
  at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1329)
  at com.instabug.library.internal.storage.cache.db.SQLiteDatabaseWrapper.query(SQLiteDatabaseWrapper.java:2)
  at com.instabug.library.internal.storage.cache.user.a.a(UserDbHelper.java:40)
  at com.instabug.library.internal.storage.cache.user.UserCacheManager.insertIfNotExists(UserCacheManager.java:2)
  at com.instabug.library.user.b.h(UserManager.java:15)
  at com.instabug.library.h.o(SessionManager.java:4)
  at com.instabug.library.h.m(SessionManager.java:7)
  at com.instabug.library.h.b(SessionManager.java:4)
  at com.instabug.library.tracking.a.onTrimMemory(InstabugActivityLifecycleListener.java:7)
  at android.app.Application.onTrimMemory(Application.java:287)
  at android.app.ActivityThread.handleTrimMemory(ActivityThread.java:6007)
  at android.app.ActivityThread.access$1100(ActivityThread.java:220)
  at android.app.ActivityThread$ApplicationThread.lambda$scheduleTrimMemory$0(ActivityThread.java:1567)
  at android.app.-$$Lambda$ActivityThread$ApplicationThread$tUGFX7CUhzB4Pg5wFd5yeqOnu38.accept(lambda:-1)
  at com.android.internal.util.function.pooled.PooledLambdaImpl.doInvoke(PooledLambdaImpl.java:271)
  at com.android.internal.util.function.pooled.PooledLambdaImpl.invoke(PooledLambdaImpl.java:195)
  at com.android.internal.util.function.pooled.OmniFunction.run(OmniFunction.java:86)
  at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1041)
  at android.view.Choreographer.doCallbacks(Choreographer.java:864)
  at android.view.Choreographer.doFrame(Choreographer.java:801)
  at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1026)
  at android.os.Handler.handleCallback(Handler.java:883)
  at android.os.Handler.dispatchMessage(Handler.java:100)
  at android.os.Looper.loop(Looper.java:214)
  at android.app.ActivityThread.main(ActivityThread.java:7397)
  at java.lang.reflect.Method.invoke(Native method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:935)
```

- [anr_2020-04-21-22-03-15-154.txt](https://github.com/Instabug/Instabug-Android/files/4528554/anr_2020-04-21-22-03-15-154.txt)

### Expected Behavior
Not freezing the app

### Actual Behavior
App may sometimes freeze

### SDK Version
9.1.0

### Android Version
N/A

### Device Model
N/A

### [Optional] Project That Reproduces the Issue
N/A

**Comment by Muhammadyoussef** [comment]
Hi @raymondctc,

A quick update: We are working hard on this issue and making sure no similar issues will arise in the future. The fix is now in our testing pipeline and expected to be released within the next two weeks.

**Comment by mzelzoghbi** [comment]
Hi @raymondctc, This should be fixed in **9.1.5**, please give it a try and let us know if you faced any issue again.
Thank You 😊
